modules/bene_promo_redirect/bene_promo_redirect.module

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * @file
+ * Bene Promo Redirect code.
+ */
+
+use Drupal\Component\Utility\UrlHelper;
+
+/**
+ * Implements hook_page_attachments().
+ */
+function bene_promo_redirect_page_attachments(array &$page) {
```

Comment:
Have you thought of writing an event subscriber that subscribes to `\Symfony\Component\HttpKernel\KernelEvents::RESPONSE`? That should let you bypass cache and conditionally redirect without Javascript. If you did this you could return a cookie that disables the redirect.

Reply:
We didn't want to mess with cookies: the functional behavior is about where people are coming from, not entirely about whether they have experienced the redirect or not. It's almost as if this is more of a "Foyer" behavior than a redirect. When you enter the site, you should always enter through this place, but if you are already in the site (google sent you in through the back door, or the chimney, or whatever), then you can wander around without having to go through the foyer.

I originally tried an event subscriber (see: https://github.com/thinkshout/bene/pull/253/commits/20dc6f7aab7b99bee8a0fad3ed127e806d7ff31a)

But without the cookie, cacheing behaviors broke things.